driver/src/driver/scheduler/system.rs

```diff
@@ -133,11 +121,41 @@ impl<'a> SystemScheduler<'a> {
     }
 }
 
-impl Scheduler for SystemScheduler<'_> {
-    fn start(&mut self) -> ! {
-        loop {
-            thread::sleep(self.run_once(SystemTime::now()));
+fn log_driver_result(batch_id: BatchId, driver_result: &DriverResult) {
+    match driver_result {
+        DriverResult::Ok => info!("Batch {} solved without error.", batch_id.0),
+        DriverResult::Retry(err) => {
+            error!("Batch {} failed with retryable error: {}", batch_id.0, err)
         }
+        DriverResult::Skip(err) => error!(
+            "Batch {} failed with unretryable error: {}",
+            batch_id.0, err
+        ),
+    }
+}
+
+impl<'a> Scheduler for SystemScheduler<'a> {
+    fn start(&mut self) -> ! {
+        crossbeam_utils::thread::scope(|scope| -> ! {
```

Comment:
Does `crossbeam` just give us a thread pool?

Reply:
It's not a threadpool. It works like `std::thread::spawn` except that it doesn't require everything to be `'static` because crossbeam guarantees that the threads won't outlive the lifetime of the captured types.

Reply:
Cool! Oh, I understand why you use it now.